tests/python_package_test/test_engine.py

```diff
@@ -2267,6 +2267,92 @@ def test_multiple_feval_train():
     assert 'decreasing_metric' in evals_result['valid_0']
 
 
+def test_params_metric_feval_callable_train():
+    # Test classification
+    X, y = load_breast_cancer(return_X_y=True)
+    params = {'verbose': -1, 'objective': 'binary', 'metric': ['binary_logloss', constant_metric]}
```

Comment:
Shall we add a case where `metric` is a single customized metric function?

Reply:
I think we can test single callable with one case either classification or regression to avoid duplicating code 

```python
# for classification
'metric': [constant_metric]
```

Reply:
I agree. Shall we do this in this PR? Thanks for the fix. This PR should now work for single callable.

Reply:
Got it, just to be consistent `classification` tests single callable from `objective`